mobile/library/swift/EnvoyCxxSwiftInterop/cxx_swift_interop.h

```diff
@@ -0,0 +1,43 @@
+#import "library/cc/bridge_utility.h"
+#import "library/cc/direct_response_testing.h"
+#import "library/cc/engine_builder.h"
+#import "library/common/data/utility.h"
+#import "library/common/extensions/filters/http/platform_bridge/c_types.h"
+#import "library/common/main_interface.h"
+#import "library/common/network/apple_platform_cert_verifier.h"
+#import "library/common/stats/utility.h"
+
+namespace Envoy {
+namespace CxxSwift {
+
+using StringVector = std::vector<std::string>;
+using StringPair = std::pair<std::string, std::string>;
+using StringPairVector = std::vector<StringPair>;
+using StringMap = absl::flat_hash_map<std::string, std::string>;
+using HeaderMatcherVector = std::vector<DirectResponseTesting::HeaderMatcher>;
+using BootstrapPtr = intptr_t;
+
+inline void string_map_set(StringMap& map, std::string key, std::string value) { map[key] = value; }
+
+inline void raw_header_map_set(Platform::RawHeaderMap& map, std::string key,
+                               std::vector<std::string> value) {
+  map[key] = value;
+}
+
+inline const DirectResponseTesting::MatchMode DirectResponseMatchModeContains =
```

Comment:
Do we perhaps want constexpr here instead of inline?
```
constexpr DirectResponseTesting::MatchMode DirectResponseMatchModeContains = DirectResponseTesting::contains;
```

Reply:
This crashes the Swift compiler.

```
1.  Apple Swift version 5.7.2 (swiftlang-5.7.2.135.5 clang-1400.0.29.51)
2.  Compiling with the current language version
3.  While evaluating request ASTLoweringRequest(Lowering AST to SIL for module Envoy)
4.  While silgen emitFunction SIL function "@$sSo5EnvoyO8CxxSwiftO31DirectResponseMatchModeContainsAB0dE7TestingO0fG0VvgZ".
 for getter for DirectResponseMatchModeContains (in module '__ObjC')
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  swift-frontend           0x0000000104bbb5b0 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 56
1  swift-frontend           0x0000000104bba5b4 llvm::sys::RunSignalHandlers() + 112
2  swift-frontend           0x0000000104bbbc34 SignalHandler(int) + 344
3  libsystem_platform.dylib 0x0000000187a9c2a4 _sigtramp + 56
4  swift-frontend           0x00000001008918a0 swift::ASTVisitor<(anonymous namespace)::StmtEmitter, void, void, void, void, void, void>::visit(swift::Stmt*) + 1636
5  swift-frontend           0x00000001008918a0 swift::ASTVisitor<(anonymous namespace)::StmtEmitter, void, void, void, void, void, void>::visit(swift::Stmt*) + 1636
6  swift-frontend           0x0000000100891fe4 swift::ASTVisitor<(anonymous namespace)::StmtEmitter, void, void, void, void, void, void>::visit(swift::Stmt*) + 3496
7  swift-frontend           0x000000010084d398 swift::Lowering::SILGenFunction::emitFunction(swift::FuncDecl*) + 724
8  swift-frontend           0x00000001007bf210 swift::Lowering::SILGenModule::emitFunctionDefinition(swift::SILDeclRef, swift::SILFunction*) + 10804
9  swift-frontend           0x00000001007c4960 swift::ASTLoweringRequest::evaluate(swift::Evaluator&, swift::ASTLoweringDescriptor) const + 6040
10 swift-frontend           0x0000000100890f10 swift::SimpleRequest<swift::ASTLoweringRequest, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule> > (swift::ASTLoweringDescriptor), (swift::RequestFlags)9>::evaluateRequest(swift::ASTLoweringRequest const&, swift::Evaluator&) + 216
11 swift-frontend           0x00000001007c76e4 llvm::Expected<swift::ASTLoweringRequest::OutputType> swift::Evaluator::getResultUncached<swift::ASTLoweringRequest>(swift::ASTLoweringRequest const&) + 628
12 swift-frontend           0x0000000100208c9c swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 7208
13 swift-frontend           0x00000001001a9294 swift::mainEntry(int, char const**) + 3940
14 dyld                     0x0000000187743e50 start + 2544
```

Reply:
Filed https://github.com/apple/swift/issues/64217